### PR TITLE
feat: support shorthand help CLI flag

### LINKED ISSUE #963: 💭 Shorthand CLI flags
Hey!

I've juste installed Rome and wanted to get the help page: my first reflex was to type `rome -h` in the terminal. Unfortunately, like all the other flags in Rome, there is no shortcut. It would've been great if there was something like `-h` for `--help`, `-v` for `--version` etc etc, for every or most of the Rome flags.

I would've love to make a PR for that but Rome uses its own argument parsing library and I don't have the time to see how it works.

### PR DESCRIPTION
**Pull request by bitpshr**
## Summary

This pull request updates the CLI to properly support shorthand flags, and adds support for a single`-h` shorthand flag that corresponds to the full `--help` flag. To date, Rome has intentionally chosen not to support shorthand CLI flags due to their ambiguity and lack of clarity, and I feel this strategy should be maintained. Still, `-h` is almost pseudo-standard, and supporting only this shorthand flag will still [allow users who _are_ expecting shorthands to exist](https://github.com/romefrontend/rome/issues/963) see the help printout as-expected, which in-turn informs them that no other shorthand flags are actually supported.

**TLDR;** Supporting a single `-h` shorthand flag for `--help` softens the Rome learning curve by providing pseudo-standard access to all other flags, and informs users that only non-shorthand flags otherwise exist.

Resolves #963

## Test Plan

```
>   ./rome test
```

### CODE REVIEW

File: internal/cli-flags/Parser.ts [hunk 1]
```diff
@@ -375,12 +371,6 @@ export default class Parser<T> {
 			consumer.markUsedProperty(key);
 		}
 
-		for (const shorthandName of this.shorthandFlags) {
-			consumer.get(shorthandName).unexpected(
-				descriptions.FLAGS.UNSUPPORTED_SHORTHANDS,
-			);
-		}
```
Comment (bitpshr) [inline internal/cli-flags/Parser.ts:382]: It felt weird to continue to explicitly raise a diagnostic when shorthand commands are encountered. In my mind, the `--help` output shows what commands are supported, which makes it clear that no shorthands exist. Passing unsupported shorthand commands (which is anything other than `-h`) should fail silently just like passing unsupported full flags.

Comment (sebmck) [reply internal/cli-flags/Parser.ts:382]: Yeah that makes sense to me. Want to remove the `UNSUPPORTED_SHORTHANDS` diagnostic description too?

File: internal/cli-flags/Parser.test.md [hunk 2]
```diff
@@ -20,7 +20,7 @@ Object {}
 
  Global Flags ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
 
-  --help, -h                        Show this help screen
```
Comment (ematipico) [inline internal/cli-flags/Parser.test.md]: Some regression, probably

Comment (bitpshr) [reply internal/cli-flags/Parser.test.md]: Good catch.

**Comment by sebmck** [comment]
Looks great! Thank you!
